### PR TITLE
chore: 🤖 Migrate `form.input` to `Hds::TextInput::Field`

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12469
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12471
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12472

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12469)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12471)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12472)

## Description
- Migrate `form/host-set/aws`, `form/host-set/azure` and `form/host-set/static` forms to use `Hds::TextInputField` instead of `form.input`
- Refactor test to use `data-test-*` attributes
- Use `InfoField` component to display type

## How to Test

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

~~- [ ] I have added before and after screenshots for UI changes~~
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: ui/admin/app/components/form/host-set/aws/index.hbs [hunk 1]
```diff
@@ -12,30 +12,31 @@
...
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
   <Hds::Form::Textarea::Field
+    name='description'
     @isOptional={{true}}
     @value={{@model.description}}
```
Comment (calcaide) [inline ui/admin/app/components/form/host-set/aws/index.hbs:38]: None blocking comment: I will suggest having the attribute `@value` before `@isOptional` to facilitate development experience.

File: ui/admin/app/components/form/host-set/aws/index.hbs [hunk 2]
```diff
@@ -50,12 +51,9 @@
     {{/if}}
   </Hds::Form::Textarea::Field>
 
-  <form.input
-    @value={{@model.type}}
-    @label={{t 'form.type.label'}}
-    readonly={{true}}
-    @disabled={{true}}
-  />
+  <InfoField @value={{@model.type}} disabled={{form.disabled}} as |F|>
```
Comment (calcaide) [inline ui/admin/app/components/form/host-set/aws/index.hbs:54]: Why the new implementation is adding a second `<InfoField>` ? in line 58 we have the previous one.

File: ui/admin/app/components/form/host-set/azure/index.hbs [hunk 3]
```diff
@@ -12,30 +12,31 @@
   as |form|
 >
 
-  <form.input
-    @name='name'
-    @type='name'
+  <Hds::Form::TextInput::Field
+    name='name'
+    @isOptional={{true}}
     @value={{@model.name}}
```
Comment (calcaide) [inline ui/admin/app/components/form/host-set/azure/index.hbs:18]: None blocking comment: I will suggest the attribute `@value` to go before `@isOptional`

Comment (dvonanderson) [reply ui/admin/app/components/form/host-set/azure/index.hbs:17]: @calcaide totally agree, quick commit with changes incoming

File: ui/admin/app/components/form/host-set/aws/index.hbs (hunk 1 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/host-set/aws/index.hbs:37]: @calcaide totally agree, quick commit with changes incoming

**Review by calcaide — APPROVED** [review]
Good migration!! :raised_hands:

**Review by cameronperera — APPROVED** [review]

**Pull request merged**